File: platform/Android/sweeteditor/src/main/java/com/qiplat/sweeteditor/EditorMetadata.java
package com.qiplat.sweeteditor;

/**
 * Editor metadata marker interface.
 * External implementations can use this interface to define custom metadata and attach it to the editor instance.
 * When used, cast to the specific subtype.
 *
 * <pre>
 * public class FileMetadata implements EditorMetadata {
 *     public final String filePath;
 *     public FileMetadata(String filePath) { this.filePath = filePath; }
 * }
 * editor.setMetadata(new FileMetadata("/a/b.cpp"));
 * FileMetadata file = (FileMetadata) editor.getMetadata();
 * </pre>
 */
public interface EditorMetadata {
}

File: platform/Apple/SweetEditor-Shared/EditorMetadata.swift
import Foundation

// MARK: - EditorMetadata

/// Editor metadata protocol.
/// External callers can implement this protocol to attach custom metadata to an editor view.
/// Use `as?` to cast it back to the concrete type when reading.
///
/// Example:
/// ```swift
/// class FileMetadata: EditorMetadata {
///     let filePath: String
///     init(filePath: String) { self.filePath = filePath }
/// }
/// editor.metadata = FileMetadata(filePath: "/a/b.cpp")
/// let file = editor.metadata as? FileMetadata
/// ```
public protocol EditorMetadata: AnyObject {}
